### PR TITLE
Recommend installing stack instead of MinGHC #75

### PR DESCRIPTION
**Pull request by snoyberg**
Pinging @ndmitchell and @3noch. I'm putting this up for a PR because I really do want this to be a discussion, not just a decision on my part. It seems to me like this is a fair assessment of what we discussed on #75, but let's make sure this correctly reflects what we all believe.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,31 @@
+# stack
```
Comment (3noch) [inline README.md]: Right at the head of MinGHC seems a bit surprising. Perhaps this could be a subsection instead?

**Comment by 3noch** [comment]
I'm certainly a fan of this in principle.

**Comment by ndmitchell** [comment]
My only concern is that I often want GHC 7.4, 7.6, 7.8 and 7.10 all installed. In contrast, with Stack, I tend to get various lts versions, and that doesn't really give me GHC versions. Maybe that's really a weakness in Stack, and there should be ghc-7.10.N as a tag, which is the most recent lts of GHC 7.10? If there was a good story there I'd be in favour.

**Comment by ndmitchell** [comment]
Also the minghc-7.8.bat script is invaluable, I don't know how to do that with stack.

**Comment by snoyberg** [comment]
I think I can answer both of those:

* `stack --resolver ghc-7.8.4` does work. We only go back as far as 7.8.4 though, so you won't be able to get 7.4 out of stack.
* I don't know the exact cmd invocation, but shouldn't something like `stack --resolver ghc-7.8.4 exec cmd` work?

FWIW, I get by when developing on Windows just fine by calling `stack ghc` and `stack runghc`.

@3noch I just pushed a commit, though I'm not thrilled with how it makes the README look. I'm OK with this or other tweaks.

**Comment by 3noch** [comment]
I was thinking something akin to

# MinGHC

## Stop: You likely want to use stack instead.....
....

**Comment by 3noch** [comment]
@snoyberg I made some edits. Feel free to change. Like I said, I'm for this change.

File: README.md [hunk 2]
```diff
@@ -1,10 +1,33 @@
 # Minimum GHC Installer [![Build Status](https://img.shields.io/travis/fpco/minghc.svg?style=flat)](https://travis-ci.org/fpco/minghc)
 
+## Wait, are you sure you don't want `stack`?
```
Comment (3noch) [inline README.md]: I `codified` `stack` to minimize ambiguity for someone very new to the Haskell ecosystem.

Comment (snoyberg) [inline README.md]: So should we go ahead with this?

**Comment by 3noch** [comment]
Any thoughts @ndmitchell?

The README is probably a nice litmus test for this change since I'm guessing most newcomers find MinGHC via haskell.org.

File: README.md [hunk 3]
```diff
@@ -1,10 +1,33 @@
...
+* Simple upgrade procedure for `stack` itself
+* Isolated environment that avoids contaminating your `PATH` variable
+
+For more information on using `stack`, please [read the `stack`
+guide](https://github.com/commercialhaskell/stack/blob/master/GUIDE.md).
+
+## MinGHC
+
+MinGHC is still a supported and active project, and makes sense for people
+looking for network-free installers for a Haskell toolchain. For more
```
Comment (ndmitchell) [inline README.md]: Or specific GHC versions including older versions.

**Comment by ndmitchell** [comment]
I've made one tweak. I also suggest we also supply a bullet point list of how to use Stack like you would minghc. Something like download binary, add to path, type stack install --ghc, run stack ghc. At the moment it's not immediately obvious how stack can do the job of minghc.

File: README.md (hunk 3 above)
Comment (3noch) [inline README.md]: Can't you tell `stack` to use specific GHC versions too?

**Comment by 3noch** [comment]
Good call. I think it'd be better to have stack host some sort of install story for Windows and just link to that.

File: README.md (hunk 3 above)
Comment (ndmitchell) [inline README.md]: Only back to 7.8

**Comment by ndmitchell** [comment]
That would be even better.

File: README.md (hunk 3 above)
Comment (3noch) [inline README.md]: But we're not uploading anything prior to that anyway?

Comment (ndmitchell) [inline README.md]: We have 7.2.2 on the web page.

Comment (snoyberg) [inline README.md]: Why not make an explicit exception in this list saying "if you need an older version of GHC, use MinGHC."

**Comment by snoyberg** [comment]
To be clear: we're talking about stack providing an NSIS-based installer for Windows in addition to the raw executable, correct? I can definitely see value in that, and it would need to just:

1. Copy stack to $APPDATA\local\bin\stack
2. Add $APPDATA\local\bin to the PATH

I'm sure there are issues we'd need to get into though about code signing, user vs global install, but something along those lines seems reasonable.

**Comment by 3noch** [comment]
Actually I was just talking about some sort of "How to" guide. But I have also thought of doing a tiny installer for stack in the past. Perhaps now's the time for that?

**Comment by snoyberg** [comment]
The installation procedure as it is today is covered on the downloads page: https://github.com/commercialhaskell/stack/wiki/Downloads#windows

**Comment by snoyberg** [comment]
And there's already an issue about an nsis installer at: https://github.com/commercialhaskell/stack/issues/613

**Comment by 3noch** [comment]
Cool!

**Comment by ndmitchell** [comment]
I was merely talking about instructions, not an installer, on which I'm ambivalent (I prefer zip files to installers). The download says how to get stack on your PATH. With MinGHC we put a working ghc on your path, which is very useful for people experimenting and not using cabal (which is pretty much all beginners to start with). What's missing is "run stack ghc or stack ghci to invoke ghc", and then stack gives the MinGHC experience.

**Comment by snoyberg** [comment]
OK, merging. We can improve the language over time, and the installer issue is already tracked upstream with stack.

**Pull request merged**